app/jobs/update_c2_proposal_job.rb

```diff
@@ -1,8 +1,10 @@
 class UpdateC2ProposalJob < ActiveJob::Base
   queue_as :default
 
-  def perform(auction_id)
+  def perform(auction_id, attributes_classname)
     auction = Auction.find(auction_id)
-    UpdateC2Proposal.new(auction).perform
+    attributes_class = attributes_classname.constantize
```

Comment:
Delayed::Job will not serialize classes, so we pass the class name as a string and constantize it.